client/v3/kubernetes/client.go

```diff
@@ -31,24 +31,20 @@ func New(cfg clientv3.Config) (*Client, error) {
 	}
 	kc := &Client{
 		Client: c,
-		kv:     clientv3.RetryKVClient(c),
 	}
-	kc.Kubernetes = kc
 	return kc, nil
 }
 
 type Client struct {
 	*clientv3.Client
-	Kubernetes Interface
-	kv         pb.KVClient
 }
 
 var _ Interface = (*Client)(nil)
 
 func (k Client) Get(ctx context.Context, key string, opts GetOptions) (resp GetResponse, err error) {
-	rangeResp, err := k.kv.Range(ctx, getRequest(key, opts.Revision))
+	rangeResp, err := k.KV.Get(ctx, key, clientv3.WithRev(opts.Revision), clientv3.WithLimit(1))
```

Comment:
minor comment, either way works, but `k.Get` is a little simpler. This comment applies to some other similar places as well. 
```suggestion
	rangeResp, err := k.Get(ctx, key, clientv3.WithRev(opts.Revision), clientv3.WithLimit(1))
```

Reply:
That leads to infinite recursion.